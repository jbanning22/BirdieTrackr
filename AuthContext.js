import {createContext} from 'react';

export const AuthContext = createContext();

// export const AuthProvider = ({children}) => {
//   const [signedIn, setSignedIn] = useState(false);

//   return (
//     <AuthContext.Provider value={{signedIn, setSignedIn}}>
//       {children}
//     </AuthContext.Provider>
//   );
// };
