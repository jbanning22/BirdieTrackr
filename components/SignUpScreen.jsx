import {
  StyleSheet,
  Text,
  View,
  TextInput,
  Button,
  TouchableOpacity,
} from 'react-native';
import React, {useState} from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {useNavigation, CommonActions} from '@react-navigation/native';
import axios from 'axios';
// import {authentication} from '../firebase/firebase-config';
// import {createUserWithEmailAndPassword} from 'firebase/auth';
// import {useDispatch} from 'react-redux';
// import {setSignedIn} from '../redux/signedInSlice';

const SignUpScreen = ({navigation}) => {
  const navigation1 = useNavigation();
  //   const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [username, setUsername] = useState('');

  const signUp = async () => {
    try {
      const signUpRes = await axios.post('http://localhost:3000/auth/signup', {
        email: email,
        password: password,
        username: username,
      });
      console.log('sign up res', signUpRes.data.access_token);
      console.log('sign up res', signUpRes.data.refresh_token);
      if (signUpRes.status === 201) {
        const access_token = await signUpRes.data.access_token;
        const refresh_token = await signUpRes.data.refresh_token;
        await AsyncStorage.setItem('token', access_token);
        await AsyncStorage.setItem('ReToken', refresh_token);
        await AsyncStorage.setItem('signedIn_status', 'allGood');
        // navigation.navigate('Scorecard', {
        //   screen: 'Scorecard',
        //   initial: true,
        // });
        // navigation1.navigate('AppStackScreen', {screen: 'Scorecard'});
        return signUpRes.data;
      } else {
        return false;
      }
    } catch (error) {
      console.log('error signing up', error);
    }
  };

  return (
    <View style={styles.box1}>
      <Text style={styles.singUpText}>Join the IDISC Community!</Text>
      <TextInput
        placeholder="Username"
        style={styles.emailInput}
        value={username}
        onChangeText={setUsername}
        clearButtonMode={'always'}
      />
      <TextInput
        placeholder="Email"
        style={styles.emailInput}
        value={email}
        onChangeText={setEmail}
        clearButtonMode={'always'}
      />
      <TextInput
        placeholder="Password"
        style={styles.loginTextInput}
        value={password}
        onChangeText={setPassword}
        clearButtonMode={'always'}
        secureTextEntry={true}
      />
      <TouchableOpacity style={styles.signUpButton}>
        <Text style={styles.textButton} onPress={signUp}>
          Sign Up
        </Text>
      </TouchableOpacity>
      <Button title="Back" onPress={() => navigation.navigate('Landing')} />
    </View>
  );
};

export default SignUpScreen;

const styles = StyleSheet.create({
  box1: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  singUpText: {
    fontSize: 28,
    fontWeight: '500',
    fontFamily: 'Helvetica',
    marginBottom: 80,
  },
  signUpButton: {
    width: 200,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#E58315',
    marginTop: 35,
    marginBottom: 15,
  },
  loginTextInput: {
    height: 50,
    width: 200,
    padding: 10,
    backgroundColor: 'white',
    // marginTop: 10,
  },
  emailInput: {
    height: 50,
    width: 200,
    padding: 10,
    backgroundColor: 'white',
    marginBottom: 15,
  },
  textButton: {
    fontSize: 18,
    color: 'white',
  },
});
