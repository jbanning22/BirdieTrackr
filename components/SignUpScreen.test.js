import React from "react";
import {render, screen, fireEvent} from "@testing-library/react-native";
import SignUpScreen from "./SignUpScreen";
import {AuthContext} from "../AuthContext";

describe("Sign Up Screen", () => {
	it("should match the snapshot", () => {
		const authContextValue = {
			signedIn: false,
			setSignedIn: jest.fn(),
		};
		const {toJSON} = render(
			<AuthContext.Provider value={authContextValue}>
				<SignUpScreen />
			</AuthContext.Provider>,
		);
		expect(toJSON()).toMatchSnapshot();
	});
	it("should render title", async () => {
		const authContextValue = {
			signedIn: false,
			setSignedIn: jest.fn(),
		};
		await render(
			<AuthContext.Provider value={authContextValue}>
				<SignUpScreen />
			</AuthContext.Provider>,
		);
		expect(screen.getByText("Join the IDISC Community!")).toBeTruthy();
	});
	it("should display sign up button", async () => {
		const authContextValue = {
			signedIn: false,
			setSignedIn: jest.fn(),
		};
		await render(
			<AuthContext.Provider value={authContextValue}>
				<SignUpScreen />
			</AuthContext.Provider>,
		);
		expect(screen.getByText("Sign Up")).toBeTruthy();
	});
	it("should navigate back to app landing", async () => {
		const authContextValue = {
			signedIn: false,
			setSignedIn: jest.fn(),
		};
		const navigation = {
			navigate: jest.fn(),
		};
		const {getByText} = await render(
			<AuthContext.Provider value={authContextValue}>
				<SignUpScreen navigation={navigation} />
			</AuthContext.Provider>,
		);
		await fireEvent.press(getByText("Back"));
		expect(navigation.navigate).toHaveBeenCalledWith("Landing");
	});
});
