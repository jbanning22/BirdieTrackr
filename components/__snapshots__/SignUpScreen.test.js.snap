// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up Screen should match the snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignSelf": "flex-start",
          "marginLeft": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View>
        <Text
          style={
            {
              "alignSelf": "center",
              "fontFamily": "Satoshi-Medium",
              "fontSize": 30,
              "fontWeight": "600",
            }
          }
        >
          Welcome to 
          <Text
            style={
              {
                "color": "#45B369",
              }
            }
          >
            BirdieTrackr
          </Text>
        </Text>
        <Text
          style={
            {
              "alignSelf": "center",
              "color": "grey",
              "fontFamily": "Satoshi-Medium",
              "fontSize": 17,
              "margin": 5,
              "textAlign": "center",
            }
          }
        >
          Create a commitment-free profile to explore products.
        </Text>
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "margin": 2,
              "width": 327,
            }
          }
        >
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Username"
            style={
              {
                "flex": 1,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "margin": 2,
              "width": 327,
            }
          }
        >
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="First Name"
            style={
              {
                "flex": 1,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "margin": 2,
              "width": 327,
            }
          }
        >
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Last Name"
            style={
              {
                "flex": 1,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "margin": 2,
              "width": 327,
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Email"
            style={
              {
                "flex": 1,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "margin": 2,
              "width": 327,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="City"
            style={
              {
                "flex": 1,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "margin": 2,
              "width": 327,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="State"
            style={
              {
                "flex": 1,
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "margin": 2,
              "width": 327,
            }
          }
        >
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              {
                "flex": 1,
              }
            }
            value=""
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "red",
              "marginTop": 10,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#2D6061",
        "borderRadius": 14,
        "height": 56,
        "justifyContent": "center",
        "margin": 15,
        "opacity": 0.4,
        "width": 327,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "Satoshi-Medium",
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#6B7280",
          "fontFamily": "Satoshi-Medium",
          "fontSize": 14,
          "paddingBottom": 5,
          "textAlign": "center",
        }
      }
    >
      By signing up, you agree to the
       
      <Text
        style={
          {
            "color": "black",
            "fontWeight": "400",
          }
        }
      >
        BirdieTrackr Terms & Conditions
      </Text>
       
      and 
      <Text
        style={
          {
            "color": "black",
            "fontWeight": "400",
          }
        }
      >
        Privacy Policy
      </Text>
      .
    </Text>
  </View>
</RCTSafeAreaView>
`;
